app/scripts/metamask-controller.js

```diff
@@ -2639,7 +2639,34 @@ export default class MetamaskController extends EventEmitter {
             this.controllerMessenger,
             'SnapController:clearSnapState',
           ),
-          getMnemonic: this.getPrimaryKeyringMnemonic.bind(this),
+          getMnemonic: async (source) => {
+            if (!source) {
+              return this.getPrimaryKeyringMnemonic();
+            }
+
+            try {
+              const mnemonic = await this.controllerMessenger.call(
+                'KeyringController:withKeyring',
+                {
+                  id: source,
```

Comment:
Is it possible to specify multiple params here? If so, we should probably specify `type: 'HD Keyring'` or whatever the actual string is

Reply:
Seems to be either `type` or `id`:

https://github.com/MetaMask/core/blob/728b167cf85d4fd8159815114e0331c9f8c6407c/packages/keyring-controller/src/KeyringController.ts#L1556-L1567

I can add a manual check for the type though.